dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/dataverse/template/ManageTemplatesPage.java

```diff
@@ -0,0 +1,247 @@
+        return "/template.xhtml?id=" + cloneTemplate.get().getId() + "&ownerId=" + dataverse.getId() + "&faces-redirect=true";
+    }
+
+    public void deleteTemplate() {
+        templateService.deleteTemplate(dataverse, selectedTemplate)
+                .onFailure(throwable -> JH.addMessage(FacesMessage.SEVERITY_FATAL, BundleUtil.getStringFromBundle("template.delete.error")))
+                .onSuccess(dataverse -> {
+                    JsfHelper.addFlashMessage(BundleUtil.getStringFromBundle("template.delete"));
+                    templatesForView.remove(selectedTemplate);
+                });
+
+    }
+
+    public void viewSelectedTemplate(Template selectedTemplate) {
+        this.selectedTemplate = selectedTemplate;
+
+        List<DatasetField> dsfForView = datasetFieldsInitializer.prepareDatasetFieldsForView(selectedTemplate.getDatasetFields());
+        mdbForView = DatasetFieldUtil.groupByBlock(dsfForView);
+    }
+
+    /**
+     * Updates dataverse regarding which templates it can use, since you can inherit templates from parent.
+     */
+    public String updateTemplatesRoot(AjaxBehaviorEvent event) throws AbortProcessingException {
+
+        if (dataverse.getOwner() != null) {
+
+            updateDefaultTemplates();
+
+            templateService.updateDataverseTemplate(dataverse, isInheritTemplatesValue())
+                    .onFailure(throwable -> Logger.getLogger(ManageTemplatesPage.class.getName()).log(Level.SEVERE, null, throwable))
+                    .onSuccess(this::setDataverse);
+
+            if (inheritTemplatesValue && dataverse.getOwner() != null) {
+                templatesForView.addAll(dataverse.getParentTemplates());
+            } else {
+                templatesForView.removeAll(dataverse.getParentTemplates());
+            }
+        }
+
+        return StringUtils.EMPTY;
+    }
+
+    // -------------------- PRIVATE --------------------
+
+    private void updateDefaultTemplates() {
+        if (isInheritTemplatesValue() && !isDataverseHasDefaultTemplate(dataverse) && isDataverseHasDefaultTemplate(dataverse.getOwner())) {
+            dataverse.setDefaultTemplate(dataverse.getOwner().getDefaultTemplate());
+        }
+
+        if (!isInheritTemplatesValue() && isDataverseHasDefaultTemplate(dataverse)) {
+            dataverse.getParentTemplates().stream()
+                    .filter(template -> template.equals(dataverse.getDefaultTemplate()))
+                    .forEach(template -> dataverse.setDefaultTemplate(null));
+        }
+    }
+
+    private boolean isDataverseHasDefaultTemplate(Dataverse dataverse) {
+        return dataverse.getDefaultTemplate() != null;
+    }
+
+    // -------------------- SETTERS --------------------
+
+    public void setTemplatesForView(List<Template> templatesForView) {
+        this.templatesForView = templatesForView;
+    }
+
+    public void setDataverse(Dataverse dataverse) {
+        this.dataverse = dataverse;
+    }
```

Comment:
Do we use this setter? It seems counterintuitive. Dataverse is set on init and I don't see when it could be set again by xhtml

Reply:
It's used in updateTemplatesRoot  but you don't need it outside of the class so I can remove it.